packages/astro/src/content/vite-plugin-content-imports.ts

```diff
@@ -156,11 +156,8 @@ export const _internal = {
 						const entryType = getEntryType(entry, contentPaths, contentEntryExts, dataEntryExts);
 						if (!COLLECTION_TYPES_TO_INVALIDATE_ON.includes(entryType)) return;
 
-						// The content config could depend on collection entries via `reference()`.
 						// Reload the config in case of changes.
-						if (entryType === 'content' || entryType === 'data') {
-							await reloadContentConfigObserver({ fs, settings, viteServer });
-						}
+						await reloadContentConfigObserver({ fs, settings, viteServer });
```

Comment:
I'm not sure why this was previously set to not reload on config changes, but this is needed now.

Reply:
Looks like it was being handled at https://github.com/withastro/astro/blob/2abc38babfe5a647882251e578f48450565b0988/packages/astro/src/content/types-generator.ts#L146-L150, but I don't know why it's split out.

Reply:
Weird. Thanks!

Reply:
Added the conditional back, and added a comment explaining where it is